cpp/src/parquet/column_writer.h

```diff
@@ -181,6 +184,10 @@ class PARQUET_EXPORT ColumnWriter {
   /// \brief The file-level writer properties
   virtual const WriterProperties* properties() = 0;
 
+  /// \brief Return KeyValueMetadata that can be used to store key-value
+  /// metadata in ColumnChunkMetaData.
+  virtual KeyValueMetadata& key_value_metadata() = 0;
```

Comment:
I'd rather not to give the freedom to get the mutable object. Instead, we can implement a similar function equivalent to https://github.com/apache/arrow/blob/main/cpp/src/parquet/file_writer.h#L163

Reply:
+1.